sos/utilities.py

```diff
@@ -192,6 +199,51 @@ def shell_out(cmd, timeout=30, chroot=None, runat=None):
                                   chroot=chroot, chdir=runat)['output']
 
 
+class AsyncReader(threading.Thread):
+    '''Used to limit command output to a given size without deadlocking
+    sos.
+
+    Takes a sizelimit value in MB, and will compile stdout from Popen into a
+    string that is limited to the given sizelimit.
+    '''
+
+    def __init__(self, channel, sizelimit):
+        super(AsyncReader, self).__init__()
+        self.chan = channel
+        self.chunksize = 2048
```

Comment:
Isn't using 4k / `4096` for `chunksize` more appropriate to match default kernel page size? Can't it improve performance?

Reply:
It might be more appropriate to match kernel page size, but I havent seen any performance gain. Any value betwwen 1k - 16k (havent tried bigger) show similar performance.

Reply:
It makes no real difference unless you are doing direct IO (and then, probably only on partial page writes where the next write hits the "other" part).

The kernel faults this stuff in in page sized units, so whether you ask for 1k, or 4k, the same data is going to be physically read in (and readahead provides another layer of insulation between physical device performance).

Reply:
(aside from syscall and context switch overheard, that is, but I don't see that being a large effect here)

Reply:
In some ways, a true ring buffer would be the ideal solution here: I'll look into how easy this is to do in Python (it's trivial in C by pointer shuffling and has the benefit that expiring entries is effectively zero-work: you just move the end pointer back, and free up some more space in the buffer).